core/controllers/editor.py

```diff
@@ -968,3 +968,27 @@ def get(self, exploration_id):
             'answers': stats_services.get_top_state_answer_stats_multi(
                 exploration_id, current_exploration.states)
         })
+
+
+class TopUnresolvedAnswersHandler(EditorHandler):
+    """Returns a list of top N unresolved answers."""
+
+    @acl_decorators.can_edit_exploration
+    def get(self):
+        """Handles GET requests for unresolved answers."""
+        try:
+            exploration_id = self.payload.get('exploration_id')
+            state_name = self.payload.get('state_name')
+        except Exception:
+            raise self.PageNotFoundException
+
+        try:
+            unresolved_answers_with_frequency = (
+                stats_services.get_top_state_unresolved_answers(
+                    exploration_id, state_name))
+        except Exception:
```

Comment:
This is a bit broad. Why would this fail?

I can understand catching issues when e.g. the state name is not valid, but the current thing is going to just shadow all errors and make debugging impossible. We should be having some sort of clean default of None or empty dict/list if the state name doesn't exist, and this should be documented in the contract of get_top_state_unresolved_answers().

Reply:
Yep, that was what I was going to do but then I looked at other controllers and they were doing try except. I will fix it.

Reply:
Done